### PR TITLE
Add the choice for an integrator of the solution

### PR DESCRIPTION
**Pull request by Ipuch**
Related to #401 
### All Submissions:

* [ ] Have you followed the guidelines in our Contributing document [docs/contribution.md]?
* [x] Have you checked to ensure there aren't other open [Pull Requests] for the same update/change?
* [x] Have you opened/linked the issue related to your pull request?
* [ ] Have you used the tag [WIP] for on-going changes, and removed it when the pull request was ready?
* [ ] When ready to merge, have you sent a comment pinging @pariterre in it?

### New Feature Submissions:

1. [] Does your submission pass the tests (if not please explain why this is intended)?
2. [ ] Did you write a proper documentation (docstrings and ReadMe)
3. [ ] Have you linted your code locally prior to submission (using the command: `black . -l120 --exclude "external/*"`)?

### Changes to Core Features:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new examples for your core changes, as applicable?
* [x] Have you written new tests for your core changes, as applicable?

### CODE REVIEW

File: bioptim/misc/enums.py [hunk 1]
```diff
@@ -103,3 +103,17 @@ class VariableType(Enum):
...
+class SolutionIntegrator(Enum):
+    """
+    Selection of integrator to use integrate function
+    """
+    DEFAULT = None
+    SCIPY_RK23 = "RK23"
+    SCIPY_RK45 = 'RK45'
+    SCIPY_DOP853 = "DOP853"
+    SCIPY_BDF = 'BDF'
+    SCIPY_LSODA = "LSODA"
```
Comment (pariterre) [inline bioptim/misc/enums.py:117]: Please give a str to Default as well, and make sure you black (don't mix "" and '')

File: bioptim/gui/plot.py [hunk 2]
```diff
@@ -197,7 +197,7 @@ def __init__(
         automatically_organize: bool = True,
         show_bounds: bool = False,
         shooting_type: Shooting = Shooting.MULTIPLE,
-        use_scipy_integrator: bool = False,
+        integrator: SolutionIntegrator = False,
```
Comment (pariterre) [inline bioptim/gui/plot.py:200]: This should not be "False" but SolutionIntegrator.DEFAULT

File: bioptim/gui/plot.py [hunk 3]
```diff
@@ -235,7 +236,8 @@ def __init__(
 
         self.t = []
         self.t_integrated = []
-        self.use_scipy_integrator = use_scipy_integrator
+        self.integrator = integrator
+        self.use_scipy_integrator = self.integrator.value is not None
```
Comment (pariterre) [inline bioptim/gui/plot.py:240]: `self.use_scipy_integrator` is a deprecated value and should be removed

File: bioptim/optimization/optimal_control_program.py [hunk 4]
```diff
@@ -762,8 +762,8 @@ def prepare_plots(
             If the ylim should fit the bounds
         shooting_type: Shooting
             What type of integration
-        use_scipy_integrator: bool
-            Use the scipy solve_ivp integrator for RungeKutta 45 instead of currently defined integrator
+        integrator: SolutionIntegrator
+            use the ode defined by OCP or use a separate integrator provided by scipy
```
Comment (pariterre) [inline bioptim/optimization/optimal_control_program.py:766]: `Use` instead of `use`

File: bioptim/optimization/solution.py [hunk 5]
```diff
@@ -509,8 +509,8 @@ def integrate(
         continuous: bool
             If the arrival value of a node should be discarded [True] or kept [False]. The value of an integrated
             arrival node and the beginning of the next one are expected to be almost equal when the problem converged
-        use_scipy_integrator: bool
-            Ignore the dynamics defined by OCP and use a separate integrator provided by scipy
+        integrator: SolutionIntegrator
+            use the ode defined by OCP or use a separate integrator provided by scipy
```
Comment (pariterre) [inline bioptim/optimization/solution.py:513]: Don't forget to capitalize the first letter of a sentence...

File: bioptim/optimization/solution.py [hunk 6]
```diff
@@ -548,9 +548,12 @@ def integrate(
         return out
 
     def __perform_integration(
-        self, shooting_type: Shooting, keep_intermediate_points: bool, continuous: bool, use_scipy_integrator: bool
-    ):
+        self, shooting_type: Shooting, keep_intermediate_points: bool, continuous: bool, integrator: SolutionIntegrator):
         n_direct_collocation = sum([nlp.ode_solver.is_direct_collocation for nlp in self.ocp.nlp])
+
+        scipy_integrator = integrator.value
+        use_scipy_integrator = scipy_integrator is not None
```
Comment (pariterre) [inline bioptim/optimization/solution.py:555]: `use_scipy_integrator` carries no information, please remove this variable

Comment (pariterre) [reply bioptim/optimization/solution.py:555]: scipy_integrator is just a copy of integrator and is therefore also useless

File: bioptim/optimization/solution.py [hunk 7]
```diff
@@ -548,9 +548,12 @@ def integrate(
...
+
         if n_direct_collocation > 0 and not use_scipy_integrator:
```
Comment (pariterre) [inline bioptim/optimization/solution.py:557]: `[...] and integrator != SolutionIntegrator.DEFAULT:`

File: tests/test_simulate.py [hunk 8]
```diff
@@ -194,15 +194,15 @@ def test_integrate(use_scipy, ode_solver):
         _ = sol.integrate(**opts)
 
     opts["keep_intermediate_points"] = True
-    if ode_solver == OdeSolver.COLLOCATION and not use_scipy:
+    if ode_solver == OdeSolver.COLLOCATION and integrator.value is None:
```
Comment (pariterre) [inline tests/test_simulate.py:197]: integrator != SolutionIntegrator.DEFAULT

File: tests/test_simulate.py [hunk 9]
```diff
@@ -194,15 +194,15 @@ def test_integrate(use_scipy, ode_solver):
...
         with pytest.raises(RuntimeError, match="Integration with direct collocation must be not continuous"):
             sol.integrate(**opts)
         return
 
     sol_integrated = sol.integrate(**opts)
     shapes = (4, 2, 2)
 
-    decimal = 5 if use_scipy else 8
+    decimal = 5 if integrator.value is not None else 8
```
Comment (pariterre) [inline tests/test_simulate.py:205]: integrator == SolutionIntegrator.DEFAULT

**Review by pariterre — CHANGES_REQUESTED** [review]
GG so far :)

File: bioptim/misc/enums.py (hunk 1 above)
Comment (Ipuch) [reply bioptim/misc/enums.py:117]: such as str() ?

File: bioptim/misc/enums.py [hunk 10]
```diff
@@ -110,7 +110,7 @@ class SolutionIntegrator(Enum):
     Selection of integrator to use integrate function
     """
 
-    DEFAULT = None
+    DEFAULT = ""
```
Comment (pariterre) [inline bioptim/misc/enums.py:113]: Default should have a value and not "" only

**Review by pariterre — CHANGES_REQUESTED** [review]
Same comment :P

**Review by pariterre — APPROVED** [review]
LGTM

**Pull request merged**